internal/provider/resource_gitlab_tag.go

```diff
@@ -0,0 +1,123 @@
+package provider
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+var _ = registerResource("gitlab_tag", func() *schema.Resource {
+	return &schema.Resource{
+		Description: "This resource allows you to create and manage GitLab tags.",
+
+		CreateContext: resourceGitlabTagCreate,
+		ReadContext:   resourceGitlabTagRead,
+		DeleteContext: resourceGitlabTagDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
```

Comment:
@armsnyder we usually try to keep the attribute names identical to the API names. In this case upstream it's `tag_name` - it's kinda redundant in this case, but should we make it `tag_name` for consistency?

Reply:
I will wait for the answer but in my opinion, it's really redundant.